Vungle/src/main/java/com/mopub/mobileads/VungleAdapterConfiguration.java

```diff
@@ -124,4 +130,19 @@ private void applyVungleNetworkSettings(Map<String, String> configuration) {
         VungleNetworkSettings.setMinSpaceForAdLoad(minSpaceLoadAd);
         VungleNetworkSettings.setAndroidIdOptOut(isAndroidIdOpted);
     }
+
+    private void refreshBidderToken() {
+        if (isComputingToken.compareAndSet(false, true)) {
```

Comment:
Am not sure why/how FB used Thread , think if you implement what have made suggestion on SDK side for availableBidTokens API to use Callable/Future and setting timeout, we can avoid any IO blocking issue, and simplify the adapter side code and not have to use Thread.

Reply:
ANR issue: https://github.com/mopub/mopub-android-mediation/commit/79de1c0f1e684fccbd3e4d5c73f2e82d6832c3fb#diff-83ab3b5b860cec7ac16b45ce9cf53813

1. Using Future has ANR risk without timeout in UI thread.
2. Not prefer to use timeout because if `availableBidTokens` cost more than 4 seconds each time we will not have the opportunity to pass one super token to MoPub. Using Thread here we make sure can pass super token when thread runs completely.

Reply:
If availableBidTokens takes longer than 4 secs how do we even know that for any request we have the latest value of bidTokens, we could be using value from the previous time the request was made. 
If this is for Mediation waterfall and I doubt that they would accept to wait for 4+ secs to just get the bidToken in their APIs, can you please confirm the behavior / time expectation for this call.
My main concern is that we dont know how long could that thread take without any timeout, and to avoid sending outdated bidToken digest, which could cause more issues.
My suggestion was to have timeout on SDK side , and simply return null if that timeout expires. Its an error case scenario and that should be acceptable.

Reply:
As we synced in the meeting, I fixed using Future and timeout on SDK side.